src/query_ctx.c

```diff
@@ -161,6 +172,9 @@ bool QueryCtx_LockForCommit(void) {
 	Graph_AcquireWriteLock(gc->g);
 	ctx->internal_exec_ctx.locked_for_commit = true;
 
+	// Changes are being committed, clear the timeout job.
+	Timeout_ClearTimeout();
+
```

Comment:
Should be the first thing we try to do right before acquiring the GIL line 149,
if we fail clearing the timeout (timeout been triggered) we should emit an exception